client/App.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import { hot } from 'react-hot-loader';
+import io from 'socket.io-client';
+import styles from './App.css';
+
+import MessageForm from './MessageForm';
+import MessageList from './MessageList';
+import UsersList from './UsersList';
+import UserForm from './UserForm';
+
+const socket = io('/');
+
+class App extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {users: [], messages: [], text: '', name: ''};
+        this.deleteMessage = this.deleteMessage.bind(this);
+    }
+
+    componentDidMount() {
+        socket.on('message', message => this.messageReceive(message));
+        socket.on('update', ({users}) => this.chatUpdate(users));
+    }
+
+    messageReceive(message) {
+        const messages = [message, ...this.state.messages];
+        this.setState({messages});
+        console.log(this.state.messages, 'msg receive');
+    }
+
+    chatUpdate(users) {
+        this.setState({users});
+    }
+
+    handleMessageSubmit(message) {
+        const messages = [message, ...this.state.messages];
+        this.setState({messages});
+        console.log(this.state.messages, 'msg state');
+        socket.emit('message', message);
+    }
+
+    handleUserSubmit(name) {
+        this.setState({name});
+        socket.emit('join', name);
+    }
+
+    deleteMessage(e) {
+        const msgId = e.target.id;
+        const userNick = e.target.from;
```

Comment:
ale skąd pomysł, że istnieje coś takiego jak `e.target.from` ? zobacz sobie czym jest `e` czy `e.target`

`e` to tylko zdarzenie kliknięcia - nie masz tam dostępu do propsów z komponentu `Message`... chyba, że je przekażesz